internal/util/util.go

```diff
@@ -39,3 +39,13 @@ func ListKeys[T any](m map[string]T) []string {
 	}
 	return keys
 }
+
+// NewMap will create a new map based upon the provided map.
+// If the provided map is `nil`, a new, empty map will be returned.
+func NewMap[K, V comparable](m map[K]V) map[K]V {
```

Comment:
why this over https://pkg.go.dev/golang.org/x/exp/maps#Copy ?

Reply:
Fair enough; I just learned about that; will do - thanks!